kotsadm/pkg/handlers/snapshots.go

```diff
@@ -542,3 +496,114 @@ func GetVeleroStatus(w http.ResponseWriter, r *http.Request) {
 	getVeleroStatusResponse.IsVeleroInstalled = true
 	JSON(w, 200, getVeleroStatusResponse)
 }
+
+type SaveSnapshotConfigRequest struct {
+	AppID         string `json:"appId"`
+	InputValue    string `json:"inputValue"`
+	InputTimeUnit string `json:"inputTimeUnit"`
+	Schedule      string `json:"schedule"`
+	AutoEnabled   bool   `json:"autoEnabled"`
+}
+
+type SaveSnapshotConfigResponse struct {
+	Success bool   `json:"success"`
+	Error   string `json:"error,omitempty"`
+}
+
+func SaveSnapshotConfig(w http.ResponseWriter, r *http.Request) {
+	if handleOptionsRequest(w, r) {
+		return
+	}
+
+	if err := requireValidSession(w, r); err != nil {
+		logger.Error(err)
+		return
+	}
+
+	responseBody := SaveSnapshotConfigResponse{}
+	requestBody := SaveSnapshotConfigRequest{}
+	if err := json.NewDecoder(r.Body).Decode(&requestBody); err != nil {
+		logger.Error(err)
+		responseBody.Error = "failed to decode request body"
+		JSON(w, 400, responseBody)
+		return
+	}
+
+	app, err := store.GetStore().GetApp(requestBody.AppID)
+	if err != nil {
+		logger.Error(err)
+		responseBody.Error = "Internal Server Error"
+		JSON(w, 500, responseBody)
```

Comment:
we've been trying to use the status codes from the `http` package
```suggestion
		JSON(w, http.StatusInternalServerError, responseBody)
```